### ISSUE TITLE
Distance calculation doesnt take advantage of the trie

### ISSUE THREAD

**Issue by OoDeLally** [jakwings/node-ternary-search-trie#3]
I just realized the distances are calculated from scratch, without taking advantage of the nature of the trie, which is by reusing the prefixes, which are computed once for all...
This makes this package kind of worthless, I'm afraid... :cold_sweat:

**Comment by iology** [comment]
Yeah, this package is for practice only, hence its very long name… 😉 

The trie structure is simple and naive. There is no auto-trie-balancing. Keys are stored as plain strings rather than unicode code points. Only one traversal order is provided.

To terminate the traversal earlier, we should use another traversal order ("pre-order"). For prefix search and Hamming Distance it is easy, and slightly harder for Levenshtein Distance.

**Comment by OoDeLally** [comment]
I meant this algorithm http://stevehanov.ca/blog/index.php?id=114

**Comment by iology** [comment]
Yes, just that. The only difference is that its trie is not ternary, but having each parent node 0-26 child nodes. And you should use an iterative algorithm to avoid stack overflow.
